cedar-policy-validator/src/types.rs

```diff
@@ -627,7 +630,163 @@ impl Type {
         }
     }
 
+    /// Does the given `PartialValue` have this validator type?
+    ///
+    /// If the `PartialValue` is a residual with not enough information to
+    /// determine conclusively that it either does or does not typecheck (i.e.,
+    /// it does typecheck for some permissible substitutions of the unknowns,
+    /// but does not typecheck for other permissible substitutions), this is
+    /// reported as an error.
+    ///
+    /// TODO(#437): Handling of `Unknown`s is not yet complete and doesn't
+    /// properly behave according to the above description, as of this writing.
+    pub(crate) fn typecheck_partial_value(
+        &self,
+        value: &PartialValue,
+        extensions: Extensions<'_>,
+    ) -> Result<bool, GetSchemaTypeError> {
+        match value {
+            PartialValue::Value(value) => self.typecheck_value(&value, extensions),
+            PartialValue::Residual(expr) => match BorrowedRestrictedExpr::new(expr) {
+                Ok(rexpr) => self.typecheck_restricted_expr(rexpr, extensions),
+                Err(_) => Ok(false), // TODO(#437): instead of just reporting typecheck fails for all nontrivial residuals, we should do something more intelligent
+            },
+        }
+    }
+
+    /// Does the given `Value` have this validator type?
+    pub(crate) fn typecheck_value(
+        &self,
+        value: &Value,
+        extensions: Extensions<'_>,
+    ) -> Result<bool, GetSchemaTypeError> {
+        match self {
+            Type::Never => Ok(false), // no value has type Never
+            Type::Primitive {
+                primitive_type: Primitive::Bool,
+            } => Ok(matches!(value, Value::Lit(Literal::Bool(_)))),
+            Type::Primitive {
+                primitive_type: Primitive::Long,
+            } => Ok(matches!(value, Value::Lit(Literal::Long(_)))),
+            Type::Primitive {
+                primitive_type: Primitive::String,
+            } => Ok(matches!(value, Value::Lit(Literal::String(_)))),
+            Type::True => match value {
+                Value::Lit(Literal::Bool(b)) => Ok(*b),
```

Comment:
I'd find this easier to read as `Value::Lit(Literal::Bool(true)) => Ok(true)`

Reply:
Actually removed this code due to thread above with Kesha